x-pack/filebeat/input/httpjson/input.go

```diff
@@ -159,6 +161,18 @@ func run(
 	return nil
 }
 
+// The Request.Tracer.Filename may have ":" when a httpjson input has cursor config
+// The MacOs Finder will treat this as path-separator and causes to show up strange filepaths.
+// This function will sanitize characters like ":" and "/" to replace them with "_" just to be
+// safe on all operating systems.
+func sanitizeFileName(name string) string {
+	id := strings.ReplaceAll(name, ":", string(filepath.Separator))
+	id = filepath.Clean(id)
+	id = strings.ReplaceAll(id, string(filepath.Separator), "_")
+
+	return id
```

Comment:
You don't need to do this (the generated code will likely be the same, but
```suggestion
	name = strings.ReplaceAll(name, ":", string(filepath.Separator))
	name = filepath.Clean(name)
	return strings.ReplaceAll(name, string(filepath.Separator), "_")
```